modules/persistence/src/services/connector/index.ts

```diff
@@ -39,11 +39,16 @@ export const db = async () => {
 
 // all docs should be inserted with the buildId for the run.
 export const insert = async (docs: any[], collection: string, buildId: ObjectId) => {
+  const timerLabel = `insert - ${collection}`;
+  console.time(timerLabel);
   const insertSession = await db();
   try {
-    return await insertSession
-      .collection(collection)
-      .insertMany(docs.map((d) => ({ ...d, build_id: buildId, created_at: buildId.getTimestamp() })));
+    const res = await insertSession.collection(collection).insertMany(
+      docs.map((d) => ({ ...d, build_id: buildId, created_at: buildId.getTimestamp() })),
+      { ordered: false }
+    );
+    console.timeEnd(timerLabel);
```

Comment:
this will only end on success. should we add these to `finally` block

Reply:
Added this to `finally` block. Also added other similar timers to end in `finally` blocks in case db errors are thrown